### ISSUE TITLE
Remove deprecated twisted.enterprise.util

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4111]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4111|
|Type|task|
|Created|2009-11-13 20:59:45Z|
|Branch|https://github.com/twisted/twisted/tree/enterprise-util-4111-2|

[twisted/enterprise/util.py](twisted/enterprise/util.py) is deprecated, let's remove it.

<details><summary>Searchable metadata</summary>

```
trac-id__4111 4111
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_enterprise_util_4111_2 branches/enterprise-util-4111-2
branch_author__thijs__itamarst thijs, itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1258145985000000 1258145985000000
changetime__1325855777000000 1325855777000000
version__None None
owner__itamar itamar
cc__itamarst cc__thijs cc__itamar
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra set owner to @thijstriemstra<br>@thijstriemstra set status to `assigned`|
|-|-|

[#4108](https://github.com/twisted/twisted/issues/4108), [#4109](https://github.com/twisted/twisted/issues/4109), [#4110](https://github.com/twisted/twisted/issues/4110) all depend on this module so this should become a master ticket.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [28082]) Branching to 'enterprise-util-4111'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [28086]) Remove deprecated t.enterprise.util and related modules, refs [#4111](https://github.com/twisted/twisted/issues/4111)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)|@khorn set owner to @khorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)<a name="note_6"></a>|@khorn set owner to @thijstriemstra|
|-|-|

Looks good.  Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Missing a news file.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [28114]) Remove references to t.enterprise.row, refs [#4111](https://github.com/twisted/twisted/issues/4111)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra commented|
|-|-|

(In [28115]) Remove row example, refs [#4111](https://github.com/twisted/twisted/issues/4111)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra removed owner|
|-|-|

I removed some more row stuff. [twisted/persisted/journal/rowjournal.py](twisted/persisted/journal/rowjournal.py) refers to t.e.row but doesnt import it. what to do with this module, deprecate it and remove the reference to t.e.row?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

  1. Please delete all of `test_enterprise` in `test_import.py`.  These tests are almost all obsolete and we should be moving towards eliminating them completely.
  1. About `rowjournal`: yes, please deprecate it and remove the reference.  sigh.  Bug itamar about what he thinks about deprecating the rest of the journal package, too.

Thanks

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_13"></a>|@thijstriemstra commented|
|-|-|

(In [28433]) Delete obsolete test_enterprise tests, refs [#4111](https://github.com/twisted/twisted/issues/4111)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra commented|
|-|-|

(In [28437]) Deprecate twisted.persisted.journal.rowjournal. Also apply some coding standards. refs [#4111](https://github.com/twisted/twisted/issues/4111)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_15"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

I deprecated the ```rowjournal``` module, but not sure if I did it correctly, but based on this session it seems I have:

```python
>>> import twisted.persisted.journal.rowjournal
__main__:1: DeprecationWarning: twisted.persisted.journal.rowjournal is deprecated as of Twisted 10.1
>>> from twisted.persisted.journal.rowjournal import INSERT
__main__:1: DeprecationWarning: twisted.persisted.journal.rowjournal is deprecated as of Twisted 10.1
>>> from twisted.persisted.journal.rowjournal import RowJournal
__main__:1: DeprecationWarning: twisted.persisted.journal.rowjournal is deprecated as of Twisted 10.1
```

The CompatibilityPolicy doesn't describe how to deprecate a module or class, as far as I can see, so if someone could add a proper description that to page, great.

Up for review.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_16"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_11):
> Bug itamar about what he thinks about deprecating the rest of the journal package, too.

I've opened [#4298](https://github.com/twisted/twisted/issues/4298) for this and assigned it to itamar.

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|TimAllen set owner to TimAllen|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_18"></a>|TimAllen set owner to @thijstriemstra|
|-|-|

I've reviewed r28437 of the branch.
 1. I note that `t.e.adbapi` is no longer being imported from `twisted/test/test_import.py` even though it isn't deprecated - is that deliberate?
 1. As exarkun noted in comment:7, it's still missing a news file (see the relevant section of the ReviewProcess page).
 1. The CompatibilityPolicy page says "Like all changes in Twisted, deprecations must be unit tested." There should be a deprecation test added to the RowJournal tests.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set owner to @itamarst|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_21"></a>|@itamarst commented|
|-|-|

(In [33325]) Branching to 'enterprise-util-4111-2'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_22"></a>|@itamarst commented|
|-|-|

  1. test_import is bogus, and apparently there's a ticket to get rid of it, so I'm not going to bother readding adbapi.
  1. Fixed.
  1. I just deleted rowjournal. Having a deprecation warning for a module that is completely broken and unusable because we deleted its dependencies doesn't make sense.

I'll submit for review after final test run.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_23"></a>|@itamarst removed owner|
|-|-|

OK, ready for review (in case you're a new reviewer, note that the branch also covers ticket [#4109](https://github.com/twisted/twisted/issues/4109)).

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/enterprise-util-4111-2

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_24"></a>|@exarkun set owner to @itamarst|
|-|-|

In addition to [#4109](https://github.com/twisted/twisted/issues/4109), this seems to cover [#4108](https://github.com/twisted/twisted/issues/4108) and [#4110](https://github.com/twisted/twisted/issues/4110) as well, right?

The news fragment for all affected tickets should be in `.removal` files.

The `twisted.enterprise.adbapi.safe` deprecation is perhaps somewhat related to this change.  We could maybe get rid of it now as well.

Please merge once `safe` is gone, or at least file a new ticket for removing it.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_25"></a>|@itamarst set status to `closed`|
|-|-|

(In [33412]) Merge enterise-util-4111-2.

Author: thijs, itamar
Review: khorn, exarkun, TimAllen
Fixes: [#4108](https://github.com/twisted/twisted/issues/4108), [#4109](https://github.com/twisted/twisted/issues/4109), [#4110](https://github.com/twisted/twisted/issues/4110), [#4111](https://github.com/twisted/twisted/issues/4111)

Remove deprecated code from the enterprise package, along with relevant documentation and examples.
